### PR TITLE
fixed readBulkString to handle RESP_NIL

### PR DESCRIPTION
**Pull request by SyedMa3**
Fixes: Issue #213 
### Changes
- Added a corner case for handling `RESP_NIL`

### Tests

- Added a test case in `tests/set_test.go` where the expected output for a command in `RESP_NIL`

### CODE REVIEW

File: core/resp.go [hunk 1]
```diff
@@ -77,6 +77,11 @@ func readBulkString(c io.ReadWriter, buf *bytes.Buffer) (string, error) {
 		return "", err
 	}
 
+	//handling RESP_NIL case
+	if len == -1 {
+		return "(nil)", nil
```
Comment (JyotinderSingh) [inline core/resp.go:82]: While this works, it might be more idiomatic to return an empty string and let callers handle the nil case.
```suggestion
		return "", nil
```

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for the fix.

Have left a few comments.

Additionally, let's handle a nil encoding case in the `Encode` method as a part of this patch.
```go
func Encode(value interface{}, isSimple bool) []byte {
    if value == nil {
        return RESP_NIL
    }
...
```

File: core/resp.go (hunk 1 above)
Comment (SyedMa3) [reply core/resp.go:82]: Currently, it is called by `(*RESParser).DecodeOne()` which directly returns the result of `readBulkString`.

If we modify the code, it will be:

```go
...
switch b {
	case '+':
		return readSimpleString(rp.c, rp.buf)
	case '-':
		return readError(rp.c, rp.buf)
	case ':':
		return readInt64(rp.c, rp.buf)
	case '$':
		resString, err := readBulkString(rp.c, rp.buf)
		if resString == "" && err == nil {
			return "(nil)", err
		}
		return resString, err
	case '*':
		return readArray(rp.c, rp.buf, rp)
	}
...
```

Will this be fine?

Comment (JyotinderSingh) [reply core/resp.go:82]: Maybe a better idea is to return `nil,nil` instead of `"", nil` for RESP_NIL case inside `readBulkString`.

Then we can change the DecodeOne call to directly return readBulkString.
```go
...
switch b {
	case '+':
		return readSimpleString(rp.c, rp.buf)
	case '-':
		return readError(rp.c, rp.buf)
	case ':':
		return readInt64(rp.c, rp.buf)
	case '$':
		return readBulkString(rp.c, rp.buf)
	case '*':
		return readArray(rp.c, rp.buf, rp)
	}
...
```

I would imagine that it would be the client's job to interpret the `nil` as `(nil)`, at the same time our codebase can expect the golang `nil` value.
```go
{
	InCmds: []string{"SET k v EX 1", "SLEEP 2", "GET k"},
	Out:    []interface{}{"OK", "OK", nil},
},
```

Comment (JyotinderSingh) [reply core/resp.go:82]: @arpitbbhayani what are your thoughts around this?

Comment (JyotinderSingh) [reply core/resp.go:82]: Resolving this for now, merging it since this fixes our currently broken tests.

**Review by JyotinderSingh — APPROVED** [review]
Thanks a lot for this fix!
